stablehlo/dialect/StablehloOps.td

```diff
@@ -1118,15 +1119,25 @@ def StableHLO_IfOp: StableHLO_Op<"if", [
 
   let results = (outs Variadic<HLO_TensorOrToken>);
 
-  let hasVerifier = 1;
+  let extraClassDeclaration = [{
+    // Method from InferTypeOpInterface interface.
+    static bool isCompatibleReturnTypes(TypeRange l, TypeRange r) {
+      if (l.size() != r.size()) return false;
+      for (auto [lt, rt] : llvm::zip(l, r))
+        if (!mlir::hlo::isCompatibleForHloTypeInference(lt, rt))
+          return false;
+      return true;
+    }
+  }];
```

Comment:
It'd be good to avoid copy/pasting these things over and over again, but let's discuss that in #145. LGTM for this pull request.

Reply:
Sure, will submit after https://github.com/openxla/stablehlo/pull/145